### PR TITLE
[NOJIRA] Improve exit logic

### PR DESCRIPTION
**Pull request by matthewdavidson**
Got rid of the `setInterval` logic in favour of worker events to improve the exit logic.

### CODE REVIEW

**Comment by georgegillams** [comment]
Is this in preparation for March 2019?

File: test/fail/verify.sh [hunk 1]
```diff
@@ -29,4 +29,4 @@ fi
 
 set -e
 
-test ! -e /test.css
+test ! -e test.css
```
Comment (matthewdavidson) [inline test/fail/verify.sh:32]: This was missed in the previous refactor - would have always "passed" as `test.css` would never exist at root path.

Comment (georgegillams) [reply test/fail/verify.sh:32]: haha wooops!

**Review by georgegillams — APPROVED** [review]
Looks 💯

**Pull request merged**